osu.Game/Rulesets/UI/HitObjectContainer.cs

```diff
@@ -31,5 +31,11 @@ protected override int Compare(Drawable x, Drawable y)
             int i = yObj.HitObject.StartTime.CompareTo(xObj.HitObject.StartTime);
             return i == 0 ? CompareReverseChildID(x, y) : i;
         }
+
+        protected override void OnChildLifetimeBoundaryCrossed(LifetimeBoundaryCrossedEvent e)
+        {
+            if (e.Kind == LifetimeBoundaryKind.End && e.Direction == LifetimeBoundaryCrossingDirection.Forward && e.Child is DrawableHitObject hitObject)
```

Comment:
Shouldn't this trigger for the backwards direction also? There are two situations which I'm focused on here:

1. The forwarding case as you have it.
2. The backward case where the code in `DrawableHitObject.Update()` should be called (and perhaps moved into `UpdateResult()`).

Reply:
I'm gonna go ahead with merging this without my above suggestion, since that should be rather handled by replay rewinding logic.